pkg/project/registry/projectrequest/delegated/delegated_test.go

```diff
@@ -0,0 +1,88 @@
+package delegated
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+	rbaclisters "k8s.io/kubernetes/pkg/client/listers/rbac/internalversion"
+
+	"github.com/go-openapi/errors"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+)
+
+func TestDelegatedWait(t *testing.T) {
```

Comment:
I'd suggest to split this to 2 functions.

Reply:
P.S. Unless I miss something. Do we really need to check 2 cases sequentially within one test method?

Reply:
Yeah, to test the unblock we want to ensure it blocked in the first place, so it makes sense to me to have this sequence in one test.